### PR TITLE
Use vue-portal for VTeleport

### LINKED ISSUE #615: Delete Teleport target on navigating away from the page
The original teleport implementation, when creating a new teleport, the TeleportTarget checked for existing teleport targets, and if the target with the same name already existed, threw an error. However, when a user navigates away from the page by, for example, entering a new address, the old teleport target is not deleted, and therefore a new teleport cannot be created. This caused issues in the CI e2e tests.

 #606 added a change to simply delete a teleport target if a target with the same name already exists. This was simply a hotfix to pass the CI, and a better solution is needed 

> I wonder if this actually makes sense as a warning or if it should throw... duplicate teleport targets seems like a programming error that we wouldn't want to accidentally let into production.

_Originally posted by @sarayourfriend in https://github.com/WordPress/openverse-frontend/pull/606#discussion_r785985438_

### LINKED ISSUE #1078: Teleport targets still causing crashes
## Description

The teleport targets are still causing crashes despite the changes in https://sentry.io/share/issue/cfcbe6bb144445a4918c77fd288516d3/

## Reproduction

Not sure how to reproduce this anymore but it does appear to be uncommon.

## Additional context

For scroll-to-top button I believe we can continue the `forceDestroy` pattern to just let the targets auto destroy on unmount instead of raising an error.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #1417: Non existent target on VTeleport mount: "skip-to-content"
## Problem
https://sentry.io/share/issue/8fc0b4ebdcb84a4b84475fc4d61a4f51/

## Description
From my guess it looks like the page loaded and attempted to teleport to a section of the page that either didn't exist or hadn't rendered yet.

The closest related issue I could find is #817

## Reproduction
I was unable to reproduce, and this appears to only be happening in Edge on Windows 10

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Hopefully:
Fixes #1078 by @ sarayourfriend
Fixes #615 by @ obulat
Fixes #1417 by @ AetherUnbound 

## Description

This PR replaces the custom implementation of Teleport with `vue-portal` library that hopefully handles the SSR better. The library hasn't been updated for some period of time, but it's probably caused by the fact that Vue 2 is not the default anymore, and Vue3 has a native implementation of `Teleport`.

## Testing Instructions

Currently, if you look at the CI Playwright tests' log, you'll see a lot of warnings like these:
```
 playwright_1  |   ✓  e2e/filters.spec.ts:145:1 › filters are updated when media type changes (11s)
playwright_1  | [WebServer] › Rendering url /search/audio?q=birds&source=freesound
playwright_1  | [WebServer] 
playwright_1  |  WARN  VTeleportTarget: duplicate name skip-to-content, deleting previous teleport
playwright_1  | [WebServer] 
playwright_1  |  WARN  VTeleportTarget: duplicate name sidebar, deleting previous teleport
playwright_1  | [WebServer] 
playwright_1  |  WARN  VTeleportTarget: duplicate name modal, deleting previous teleport
```
The CI logs for this PR shouldn't have warnings about deleting portal targets, and the tests should not fail due to crashes.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
The [SSR docs for Vue Portal](https://portal-vue.linusb.org/guide/SSR.html) suggest disabling it on the server and only running it on the client side using `<no-ssr>`. 

I'm all for replacing a custom implementation with a tried and tested library (approving for that) but the docs seem to suggest that our issue with SSR wouldn't go away after the switch.

File: src/components/VContentSwitcher/VSearchTypeButton.vue [hunk 1]
```diff
@@ -13,8 +13,8 @@
   >
     <VIcon :icon-path="icon" />
     <span
-      class="md:block md:truncate md:text-left md:ms-2"
-      :class="isHeaderScrolled ? 'hidden' : 'block truncate text-left ms-2'"
+      class="md:block md:truncate md:ms-2 md:text-start"
+      :class="isHeaderScrolled ? 'hidden' : 'block truncate ms-2 text-start'"
```
Comment (sarayourfriend) [inline src/components/VContentSwitcher/VSearchTypeButton.vue:17]: Good catch, surprised the change didn't require updating snapshot tests though :thinking:

**Review by sarayourfriend — APPROVED** [review]
LGTM! I poked through the `portal-vue` code to see what they had to do and it's a bit more involved than the naive approach our implementation had :sweat_smile:

**Comment by obulat** [comment]
> The [SSR docs for Vue Portal](https://portal-vue.linusb.org/guide/SSR.html) suggest disabling it on the server and only running it on the client side using `<no-ssr>`.
> 
> I'm all for replacing a custom implementation with a tried and tested library (approving for that) but the docs seem to suggest that our issue with SSR wouldn't go away after the switch.

@dhruvkb, I believe this part of docs refers to the specific (older than Nuxt) implementation, the SSR for Vue. The library also has a Nuxt module, and it works pretty well. At least, I didn't notice errors.

File: src/components/VContentSwitcher/VSearchTypeButton.vue (hunk 1 above)
Comment (obulat) [reply src/components/VContentSwitcher/VSearchTypeButton.vue:17]: I caught myself using `left` and `right` classes a couple of days ago 🙈

**Pull request merged**